rust/src/main.rs

```diff
@@ -42,22 +42,22 @@ impl EvTag {
         let object = odb.read(object.id())?;
         let contentbuf = object.data();
         let header = format!("{} {}", object.kind().str(), contentbuf.len());
-        hash.write(header.as_bytes()).unwrap();
+        hash.write_all(header.as_bytes()).unwrap();
```

Comment:
Oh, that's a useful catch indeed, though I'd be surprised if any of the hashers actually did partial writes.